tests/Helpers/JWKFetcherTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Auth0\Tests\Helpers;
+
+use Auth0\SDK\Helpers\JWKFetcher;
+use josegonzalez\Dotenv\Loader;
+use GuzzleHttp\Exception\RequestException;
+
+/**
+ * Class JWKFetcherTest
+ *
+ * @package Auth0\SDK\Helpers
+ */
+class JWKFetcherTest extends \PHPUnit_Framework_TestCase
+{
+
+    /**
+     * Issuer for JWKS URL.
+     *
+     * @var string
+     */
+    public static $issuer;
+
+    /**
+     * Setup before class.
+     *
+     * @return void
+     */
+    public static function setUpBeforeClass()
+    {
+        $env_path = '.env';
+        if (file_exists($env_path)) {
+            $loader = new Loader($env_path);
+            $loader->parse()->putenv(true);
+        }
+
+        self::$issuer = 'https://'.getenv('DOMAIN').'/';
+        parent::setUpBeforeClass();
+    }
+
+    /**
+     * Test that a valid JWKS is fetched.
+     *
+     * @return void
+     */
+    public function testFetchJwks()
+    {
+        $fetcher = new JWKFetcher();
+        $keys    = $fetcher->fetchKeys( self::$issuer );
+
+        $this->assertNotEmpty( $keys );
+
+        $pem_parts = $this->getPemFromKeys( $keys );
+
+        $this->assertEquals( '-----BEGIN CERTIFICATE-----', $pem_parts[0] );
+        $this->assertEquals( '-----END CERTIFICATE-----', $pem_parts[count($pem_parts) - 2] );
+    }
+
+    /**
+     * Test that a valid JWKS is fetched with an empty path passed.
+     *
+     * @return void
+     */
+    public function testFetchJwksFullUrl()
+    {
+        $fetcher = new JWKFetcher();
+        $keys    = $fetcher->fetchKeys( self::$issuer.'.well-known/jwks.json', null );
```

Comment:
Looking at this line again. Why would a user want to call this method like this? As a dev I'd like to avoid having to set a explicit null when I'm already passing the URL in the first param. More questions raise:

- Would it make sense to have this "jwks_path" customization on a separate method, maybe as part of the constructor of the JWKFetcher instance?